pkg/reconciler/pipeline/watcher.go

```diff
@@ -0,0 +1,135 @@
+package pipeline
+
+import (
+	"context"
+	"fmt"
+
+	appv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+)
+
+func setResourceStatus(c client.Client, pipeline *dfv1.Pipeline) error {
```

Comment:
Can we move it to `pipelineReconciler` as a function `checkChildrenResourceStatus()`?